src/Mvc/Mvc.RazorPages/src/ApplicationModels/DefaultPageApplicationModelProvider.cs

```diff
@@ -98,16 +98,21 @@ public void OnProvidersExecuted(PageApplicationModelProviderContext context)
 
             // Now we want figure out which type is the handler type.
             TypeInfo handlerType;
+            var pageTypeAttributes = pageTypeInfo.GetCustomAttributes(inherit: true);
+            object[] handlerTypeAttributes;
             if (modelProperty.PropertyType.IsDefined(typeof(PageModelAttribute), inherit: true))
             {
                 handlerType = modelTypeInfo;
+
+                // If a PageModel is specified, combine the attributes specified on the Page and the Model type.
+                handlerTypeAttributes = modelTypeInfo.GetCustomAttributes(inherit: true).Concat(pageTypeAttributes).ToArray();
```

Comment:
Does it make sense to have the model type attributes first or the page type first? I can't really think of a good way to decide this strongly.

Reply:
I wanted the model (code) to be treated as being more significant than the page (code-gen). It also keeps the semantics closer to 2.x if you happened to have attributes on a custom Page base type. 

As a user, we really would recommend being explicit if the ordering is significant.

Reply:
Makes sense, would be nice to have a comment.